igcollect/sfp_dom.py

```diff
@@ -0,0 +1,353 @@
+            'high_warning': SFF_A2_TEMP_HWARN,
+            'low_critical': SFF_A2_TEMP_LALRM,
+            'high_critical': SFF_A2_TEMP_HALRM,
+        },
+        'bias_current_lane1': {
+            'current': SFF_A2_BIAS,
+            'low_warning': SFF_A2_BIAS_LWARN,
+            'high_warning': SFF_A2_BIAS_HWARN,
+            'low_critical': SFF_A2_BIAS_LALRM,
+            'high_critical': SFF_A2_BIAS_HALRM,
+        },
+        'sfp_voltage': {
+            'current': SFF_A2_VCC,
+            'low_warning': SFF_A2_VCC_LWARN,
+            'high_warning': SFF_A2_VCC_HWARN,
+            'low_critical': SFF_A2_VCC_LALRM,
+            'high_critical': SFF_A2_VCC_HALRM,
+        },
+        'tx_power_lane1': {
+            'current': SFF_A2_TX_PWR,
+            'low_warning': SFF_A2_TX_PWR_LWARN,
+            'high_warning': SFF_A2_TX_PWR_HWARN,
+            'low_critical': SFF_A2_TX_PWR_LALRM,
+            'high_critical': SFF_A2_TX_PWR_HALRM,
+        },
+        'rx_power_lane1': {
+            'current': SFF_A2_RX_PWR,
+            'low_warning': SFF_A2_RX_PWR_LWARN,
+            'high_warning': SFF_A2_RX_PWR_HWARN,
+            'low_critical': SFF_A2_RX_PWR_LALRM,
+            'high_critical': SFF_A2_RX_PWR_HALRM,
+        },
+    }
+
+    def __init__(self, data, interface):
+        # Check if SFP data is externally calibrated. We don't support it.
+        external_calibration = data[SFF_A0_DOM] & SFF_A0_DOM_EXTCAL
+        if external_calibration:
+            raise UnsupportedSFPException(
+                'SFPs calibrated externally are not supported'
+            )
+
+        # We finish the object creation only if the SFP is internally
+        # calibrated.
+        self._data = data
+        self.interface = interface
+        self.metrics = {
+            'temperature': {
+                'current': None,
+                'low_warning': None,
+                'high_warning': None,
+                'low_critical': None,
+                'high_critical': None,
+            },
+            'bias_current_lane1': {
+                'current': None,
+                'low_warning': None,
+                'high_warning': None,
+                'low_critical': None,
+                'high_critical': None,
+            },
+            'sfp_voltage': {
+                'current': None,
+                'low_warning': None,
+                'high_warning': None,
+                'low_critical': None,
+                'high_critical': None,
+            },
+            'tx_power_lane1': {
+                'current': None,
+                'low_warning': None,
+                'high_warning': None,
+                'low_critical': None,
+                'high_critical': None,
+            },
+            'rx_power_lane1': {
+                'current': None,
+                'low_warning': None,
+                'high_warning': None,
+                'low_critical': None,
+                'high_critical': None,
+            },
+        }
```

Comment:
To reduce some points to touch when the mappings changes, I would make this here dynamic:
```suggestion
        self.metrics = {cat: {metric: None for metric, _ in m_offs.items()} for cat, m_offs in OFFSET_MAP.items()}
```

Reply:
Same as above. I will address this in the new PR that I will submit.